### PR TITLE
dev-db/mongodb: bump to 3.6.1

### PR DESCRIPTION
**Pull request by hydrapolic**

### CODE REVIEW

**Comment by ultrabug** [comment]
Thanks mate, I'm testing this right now.

But as I told you, I'll be tree cleaning old versions before merging this. I'd like to have only two majors supported on portage.

**Comment by hydrapolic** [comment]
Thanks.

It would be nice to keep 3.2 for a little while :)

**Comment by hydrapolic** [comment]
That said,I'm ok with dropping 2.6/3.0.

**Comment by ultrabug** [comment]
Please make sure that the `install_compass` is NOT installed because it's misleading and not supporting gentoo.

Also, since I have the `splitdebug` FEATURE on, it leads to a 14G tmp directory! I wonder if a RESTRICT="splitdebug" would prevent this for this package... but that's my user's choice so I guess it's okay.

**Comment by hydrapolic** [comment]
Ok I'll check again on compass, thanks for noticing. About the splitdebug, should we warn the users that it takes some space? Well I can imagine that in some cases you'll run this in production if you debug some obscure crashes that are hard to reproduce in testing environment.

**Comment by hydrapolic** [comment]
@ultrabug, removed compass, bumped required versions (gcc,scons), added python deps (pyyaml,typing), also removed bundled libs. Tests pass on amd64, please give it a try ;)

I've tried to unbundle wiredtiger, but no luck so far. Also using system-icu causes the tests to fail.

File: app-admin/mongo-tools/mongo-tools-3.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	# build pie to avoid text relocations wrt #582854
+	# skip on ppc64 wrt #610984
+	if ! use ppc64; then
+		sed -i 's/\(go build\)/\1 -buildmode=pie/g' build.sh || die
+	fi
+
+	# ensure we use bash wrt #582906
+	sed -i 's@/bin/sh@/bin/bash@g' build.sh || die
+
+	default
```
Comment (SoapGentoo) [inline app-admin/mongo-tools/mongo-tools-3.6.0.ebuild]: In general, prefer to call `default` first thing in `src_prepare`

File: app-admin/mongo-tools/mongo-tools-3.6.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils
```
Comment (SoapGentoo) [inline app-admin/mongo-tools/mongo-tools-3.6.0.ebuild]: do you still need this?

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	dev-libs/snowball-stemmer
+	net-libs/libpcap
+	>=sys-libs/zlib-1.2.8:=
+	mms-agent? ( app-admin/mms-agent )
+	ssl? (
+		!libressl? ( >=dev-libs/openssl-1.0.1g:0= )
+		libressl? ( dev-libs/libressl:0= )
+	)"
+DEPEND="${RDEPEND}
+	dev-python/cheetah
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `[${PYTHON_USEDEP}]`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+	dev-python/pyyaml
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `[${PYTHON_USEDEP}]`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+	dev-python/typing
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `[${PYTHON_USEDEP}]`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+	=dev-lang/python-2*
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `${PYTHON_DEPS}` always, never depend on naked python impls

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+	<dev-util/scons-3
+	>=sys-devel/gcc-5.4.0:*
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: pointless dependency, remove

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+	sys-libs/ncurses
+	sys-libs/readline
+	debug? ( dev-util/valgrind )
+	kerberos? ( dev-libs/cyrus-sasl[kerberos] )
+	test? (
+		dev-python/pymongo
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `[${PYTHON_USEDEP}]`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+	if use debug; then
+		scons_opts+=( --dbg=on )
+	fi
+
+	if use prefix; then
+		scons_opts+=(
+			--cpppath="${EPREFIX}/usr/include"
+			--libpath="${EPREFIX}/usr/$(get_libdir)"
+		)
+	fi
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: I currently don't see why this is in a prefix conditional, maybe add it to the global option array

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+	enewgroup mongodb
+	enewuser mongodb -1 -1 /var/lib/${PN} mongodb
+
+	# Maintainer notes
+	#
+	# --use-system-icu fails tests
+	# --use-system-tcmalloc is strongly NOT recommended:
+	# https://www.mongodb.org/about/contributors/tutorial/build-mongodb-from-source/
+
+	scons_opts=(
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: do all the `scons_opts` mucking `src_configure`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+		--use-system-yaml
+		--use-system-zlib
+	)
+
+	# wiredtiger not supported on 32bit platforms #572166
+	use x86 && scons_opts+=( --wiredtiger=off )
+
+	if use debug; then
+		scons_opts+=( --dbg=on )
+	fi
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: use the more concise syntax: `use debug && scons_opts+=( --dbg=on )`

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+	default
+}
+
+src_compile() {
+	# respect mongoDB upstream's basic recommendations
+	# see bug #536688 and #526114
+	if ! use debug; then
+		filter-flags '-m*'
+		filter-flags '-O?'
+	fi
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: do this in `src_configure` after the `scons_opts` mucking

File: dev-db/mongodb/mongodb-3.6.0.ebuild [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+	# wrt bug #461466
+	if [[ "$(get_libdir)" == "lib64" ]]; then
+		rmdir "${ED}"/usr/lib/ &>/dev/null
+	fi
+}
+
+# FEATURES="test -usersandbox" emerge dev-db/mongodb
+src_test() {
+	"${EPYTHON}" ./buildscripts/resmoke.py --dbpathPrefix=test --suites core --jobs=$(makeopts_jobs) || die "Tests failed"
+}
```
Comment (SoapGentoo) [inline dev-db/mongodb/mongodb-3.6.0.ebuild]: `src_test` always between `src_compile` and `src_install`

**Comment by ultrabug** [comment]
Thanks @hydrapolic  mate, please address the review of @SoapGentoo and directly bump to 3.6.1 so I can test as well afterwards

**Comment by hydrapolic** [comment]
Yes I will,but I'm taking a break until the very first days of 2018 :) thanks @SoapGentoo for all the reviews!

**Comment by ultrabug** [comment]
np @hydrapolic, happy new year ;)

Meanwhile, I'll start cleaning 2.6 and 3.0 versions from tree

**Comment by hydrapolic** [comment]
@ultrabug, pushed up version 3.6.1, test pass on amd64, please give it a try. Thank you!

**Comment by hydrapolic** [comment]
And happy new year to you too!

**Comment by ultrabug** [comment]
Thank you mate, testing now.

**Comment by ultrabug** [comment]
Very good mate, works great for me! Thanks a lot 👍

**Comment by hydrapolic** [comment]
Thank you for the help guys!

**Comment by cjwijtmans** [comment]
any chance yet of getting mongodb/scons to python3? The only packagaes on my server on python2 atm.

**Comment by hydrapolic** [comment]
This is currently a mongodb limitation,please ask for python3 support upstream. We will be more than happy to add py3 support in gentoo then :)

**Pull request closed without merging**